src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -892,6 +895,42 @@ private String getNearestClassFrameName() {
         return frame.getFrameName();
     }
 
+    /**
+     * Checks if the token is a Lambda parameter.
+     * @param ast the {@code DetailAST} value of the token to be checked
+     * @return true if the token is a Lambda parameter
+     */
+    private static boolean isLambdaParameter(DetailAST ast) {
+        DetailAST parent;
+        for (parent = ast.getParent(); parent != null;
+             parent = parent.getParent()) {
+            if (parent.getType() == TokenTypes.LAMBDA) {
+                break;
+            }
+        }
+        final boolean isLambdaParameter;
+        if (parent == null) {
+            isLambdaParameter = false;
+        }
+        else if (ast.getType() == TokenTypes.PARAMETER_DEF) {
+            isLambdaParameter = true;
+        }
+        else {
+            final DetailAST lambdaParameters = parent.findFirstToken(TokenTypes.PARAMETERS);
+            if (lambdaParameters == null) {
+                isLambdaParameter = parent.getFirstChild().equals(ast);
+            }
+            else {
+                isLambdaParameter = TokenUtils.findFirstTokenByPredicate(lambdaParameters,
+                    paramDef -> {
+                        final DetailAST param = paramDef.findFirstToken(TokenTypes.IDENT);
+                        return param != null && param.equals(ast);
```

Comment:
I would have to sit down and look at the code in more detail to tell, but IMO, it comes down to if you are comparing the same node or are comparing the same text.
If you are looking for the same node, which might be a rare occurrence in CS, than you should use `==`.
If you are looking for the same text, than you should use `getText().equals(...)`.

`DetailAST.equals` is actually coming from `antlr.BaseAST` which I would say is not safe for usage in CS since we don't override it in our `DetailAST` class and may miss out on `DetailAST`'s custom fields.
@romani What do you feel about overriding `DetailAST`'s `equals` and `hashCode` to work with our implementation?